crates/relayer/src/main.rs

```diff
@@ -174,29 +236,93 @@ impl RelayerService {
         // Create blockchain client from relayer config
         let transports = self.create_client()?;
 
-        // Clone mock relayer for the async task
+        // Clone mock relayer and metrics for the async task
         let mock_relayer = self.mock_relayer.clone();
+        let metrics = self.metrics.clone();
 
         let handle = async move {
             while let Some((request, res_tx)) = rx.recv().await {
+                // Record queue depth (items waiting to be processed)
+                // rx.recv() has already removed the current item, so rx.len() is correct
+                if let Some(ref metrics) = metrics {
+                    metrics.set_queue_depth(rx.len() as i64);
+                }
+
                 // Check if this is a mock-relayer request
                 if request.protocol == protocols::mock_relayer::NAME {
+                    // Clone and normalize protocol name to prevent cardinality explosion
+                    let protocol_str = request.protocol.clone();
+                    let protocol_name = normalize_protocol_name(&protocol_str);
```

Comment:
no need to normalize the protocol name here as two lines before you already match it at: `                if request.protocol == protocols::mock_relayer::NAME {`. If you want to have it for consistency reasons, please remove the unnecessary clone as mentioned above, too.